spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -598,7 +598,8 @@ private static String sqlString(org.apache.iceberg.expressions.Literal<?> lit) {
       if (lit.value() instanceof String) {
         return "'" + lit.value() + "'";
       } else if (lit.value() instanceof ByteBuffer) {
-        throw new IllegalArgumentException("Cannot convert bytes to SQL literal: " + lit);
+        // produce a literal value that Spark can parse
+        return lit.toString();
```

Comment:
It is fine to update `BinaryLiteral.toString` and `FixedLiteral.toString`, but this should not assume that the `toString` representation of a literal can be parsed by Spark. Instead, this should produce the correct form here, just like it does for `String` literals.

Reply:
If we implement `BinaryLiteral.toString` and `FixedLiteral.toString`, then we don't need to do it separately for ByteBuffer

Reply:
No, we should not rely on `toString` using a representation that works for Spark. Instead, this should convert the literal for Spark here. We may change the toString representation in `Literal` and we don't want this to depend on that method. It's fine for numbers, but there are different representations for binary that are equally valid. This is where we customize the one that Spark uses.

Reply:
Okay, got it